Content.Server/GameObjects/Components/Nutrition/CigSmokingComponent.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System.Threading.Tasks;
+using Content.Server.GameObjects.Components.Items.Clothing;
+using Content.Server.GameObjects.Components.Items.Storage;
+using Content.Shared.GameObjects.Components.Nutrition;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Log;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Nutrition
+{
+    [RegisterComponent]
+    public class CigSmokingComponent : SharedSmokingComponent, IInteractUsing
+    {
+        private SmokingStates _currentState;
+        private ItemComponent _item;
+
+        [ViewVariables]
+        public SmokingStates CurrentState
+        {
+            get => _currentState;
+            private set
+            {
+                _currentState = value;
+                if (_item == null
+                    || !_item.Owner.TryGetComponent<SpriteComponent>(out SpriteComponent spriteComponent))
+                {
+                    Logger.Warning("No sprite component found for _item");
+                    return;
+                }
+                else
+                {
+                    _item.Owner.TryGetComponent(out ClothingComponent clothing);
+                    switch (_currentState)
+                    {
+                        case SmokingStates.Lit:
+                            _item.EquippedPrefix = "lit";
+                            clothing.ClothingEquippedPrefix = "lit";
+                            spriteComponent.LayerSetState(0, "lit-icon");
+                            break;
+                        default:
+                            _item.EquippedPrefix = "unlit";
+                            clothing.ClothingEquippedPrefix = "unlit";
+                            spriteComponent.LayerSetState(0, "icon");
+                            break;
```

Comment:
Why are you setting these on the server? We have client-side visualizers for a reason.

Reply:
It was the only way to get the icon of the item to show as animated. Perhaps changing netsync will solve this problem.

Reply:
It should be false regardless because currently it sends the data every tick (which is... yeah).

There's animationtracks client-side if you need them, try sussing out airlocks.